[vcpkg] Add platform support for Solaris and illumos

trisk opened the pull request:
This add support for Solaris and illumos. Tested building several ports on illumos/amd64 (OmniOS with clang 18 from pkgsrc). Will be submitting fixes for ports that need them as separate PRs.
Depends on: https://github.com/microsoft/vcpkg-tool/pull/1684

trisk committed:
[vcpkg] Add platform support for Solaris and illumos

BillyONeal commented on scripts/bootstrap.sh:
```diff
@@ -108,6 +109,12 @@ if [ "$UNAME" = "OpenBSD" ]; then
+if [ "$UNAME" = "SunOS" ]; then
+    [ -x /usr/bin/gtar ] && TAR=/usr/bin/gtar
+    [ -x /usr/gnu/bin/tar ] && TAR=/usr/gnu/bin/tar
+    vcpkgUseSystem="ON"
```
I think the block in `vcpkgCheckRepoTool` explaining how to get the bits needs a line to tell the user how to get those dependencies

trisk replied:
Looks like the `vcpkgCheckRepoTool` logic for `vcpkgSkipDependencyChecks` was inverted; it looks for `vcpkgSkipDependencyChecks=ON` before it performs checks. The default is `OFF`, while passing `-skipDependencyChecks` sets it to `ON`.

BillyONeal commented on scripts/bootstrap.sh:
```diff
@@ -22,7 +22,7 @@ do
         echo "Warning: -buildTests no longer has any effect; ignored."
     elif [ "$var" = "-skipDependencyChecks" ]; then
-        vcpkgSkipDependencyChecks="OFF"
+        vcpkgSkipDependencyChecks="ON"
```
![facepalm](https://github.com/user-attachments/assets/c2f35ebb-16b8-4274-9fa5-e111e02bde3a)

BillyONeal replied:
https://github.com/microsoft/vcpkg/pull/46216/  Oh, I see, it "works" because two bugs happen together

BillyONeal commented:
> Interestingly, there is a new `x64_osx` test failure in the newly added `tar` check, suggesting it does not actually support archives with pax extended headers either. We normally use a prebuilt `vcpkg-tool` there, but I wonder what happens with the real release tarball.

```
vcpkg@CPPMAC-ARM64-05 test % curl -L -o test.tar.gz https://github.com/microsoft/vcpkg-tool/archive/refs/tags/2025-06-02.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 2833k    0 2833k    0     0  2860k      0 --:--:-- --:--:-- --:--:-- 5392k
vcpkg@CPPMAC-ARM64-05 test % tar xf test.tar.gz
vcpkg@CPPMAC-ARM64-05 test % ls
test.tar.gz             vcpkg-tool-2025-06-02
vcpkg@CPPMAC-ARM64-05 test % tar --version
bsdtar 3.5.3 - libarchive 3.5.3 zlib/1.2.12 liblzma/5.4.3 bz2lib/1.0.8
vcpkg@CPPMAC-ARM64-05 test %
```

Something about the test doesn't seem to match up with 'reality' of extracting our tool sources?

trisk commented:
> > Interestingly, there is a new `x64_osx` test failure in the newly added `tar` check, suggesting it does not actually support archives with pax extended headers either. We normally use a prebuilt `vcpkg-tool` there, but I wonder what happens with the real release tarball.
> 
> ```
> vcpkg@CPPMAC-ARM64-05 test % curl -L -o test.tar.gz https://github.com/microsoft/vcpkg-tool/archive/refs/tags/2025-06-02.tar.gz
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 100 2833k    0 2833k    0     0  2860k      0 --:--:-- --:--:-- --:--:-- 5392k
> vcpkg@CPPMAC-ARM64-05 test % tar xf test.tar.gz
> vcpkg@CPPMAC-ARM64-05 test % ls
> test.tar.gz             vcpkg-tool-2025-06-02
> vcpkg@CPPMAC-ARM64-05 test % tar --version
> bsdtar 3.5.3 - libarchive 3.5.3 zlib/1.2.12 liblzma/5.4.3 bz2lib/1.0.8
> vcpkg@CPPMAC-ARM64-05 test %
> ```
> 
> Something about the test doesn't seem to match up with 'reality' of extracting our tool sources?

Updated the test to be libarchive compatible (and also search for `bsdtar` or `star` as shipped in pkgsrc), and fixed the error message from `bootstrap.sh` so it refers to `tar` instead of the last failed command.

BillyONeal reviewed (commented):
The added binary file makes me nervous but not an over my dead body thing yet.... what does this comment mean?

![image](https://github.com/user-attachments/assets/4179379c-e88c-490f-89e0-637b6d277dcd)

BillyONeal commented:
/azp run

BillyONeal commented:
The build failures were fixed by #46016 so we will merge through them if this change is otherwise OK

BillyONeal commented:
> I think the zip archive should work fine. Another option is removing the first block from the tarball that contains the extra header with `tail -c +1025`, if we can expect that to be present consistently.

I think the zip is better because we don't really know how / if that metadata is setup

trisk committed:
[bootstrap] Add dependency instructions for Solaris and illumos

trisk committed:
[bootstrap] Use source zip instead of tarball to avoid pax headers

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into HEAD

# Conflicts:
#	scripts/vcpkg-tool-metadata.txt

BillyONeal reviewed (approved):
Thanks!
